felix/rules/policy.go

```diff
@@ -142,6 +142,50 @@ func filterNets(mixedCIDRs []string, ipVersion uint8) (filtered []string, filter
 	return
 }
 
+// isUniversalCIDR returns true if the CIDR represents "all addresses" for the given IP version.
+// This is used to detect problematic negated matches that would create logical contradictions.
+func isUniversalCIDR(cidr string, ipVersion uint8) bool {
+	if ipVersion == 4 {
+		return cidr == "0.0.0.0/0"
+	} else if ipVersion == 6 {
+		return cidr == "::/0"
+	}
+	return false
+}
+
+// filterNetsAndValidateNegation filters CIDRs by IP version and validates that negated universal
+// CIDRs are not present, which would create logical contradictions in iptables rules.
+func filterNetsAndValidateNegation(mixedCIDRs []string, ipVersion uint8, isNegated bool) (filtered []string, filteredAll bool) {
```

Comment:
We added this new `filterNetsAndValidateNegation()` function with an `isNegated` argument but it's only being called for the negated rules. Did you mean to replace `filterNets()` with this and use `isNegated` to differentiate the behaviour?

We might just keep calling it `filterNets()` then, and replace it with this function. What do you think?